mne/viz/eyetracking/tests/test_heatmap.py

```diff
@@ -0,0 +1,42 @@
+# Authors: Scott Huberty <seh33@uw.edu>
+#
+# License: Simplified BSD
+
+import pytest
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import mne
+
+
+@pytest.mark.parametrize(
+    "axes",
+    [(None), (True)],
+)
+def test_plot_heatmap(axes):
+    """Test plot_gaze."""
+    # Create a toy epochs instance
+    info = info = mne.create_info(
+        ch_names=["xpos", "ypos"], sfreq=100, ch_types="eyegaze"
+    )
+    # here we pretend that the subject was looking at the center of the screen
+    # we limit the gaze data between 860-1060px horizontally and 440-640px vertically
+    data = np.vstack([np.full((1, 100), 1920 / 2), np.full((1, 100), 1080 / 2)])
+    epochs = mne.EpochsArray(data[None, ...], info)
+    epochs.info["chs"][0]["loc"][4] = -1
+    epochs.info["chs"][1]["loc"][4] = 1
+
+    if axes:
+        axes = plt.subplot()
+    fig = mne.viz.eyetracking.plot_gaze(
+        epochs, width=1920, height=1080, axes=axes, cmap="Greys"
+    )
+    img = fig.axes[0].images[0].get_array()
+    # the pixels in the center of canvas
+    assert 960 in np.where(img)[1]
+    assert np.isclose(np.min(np.where(img)[1]), 860)
+    assert np.isclose(np.max(np.where(img)[1]), 1060)
+    assert 540 in np.where(img)[0]
+    assert np.isclose(np.min(np.where(img)[0]), 440)
+    assert np.isclose(np.max(np.where(img)[0]), 640)
```

Comment:
what are these `isclose` assertions doing?

```suggestion
    assert width / 2 in np.where(img)[1]
    assert np.isclose(np.min(np.where(img)[1]), 860)
    assert np.isclose(np.max(np.where(img)[1]), 1060)
    assert height / 2 in np.where(img)[0]
    assert np.isclose(np.min(np.where(img)[0]), 440)
    assert np.isclose(np.max(np.where(img)[0]), 640)
```

Reply:
FYI I changed the test so it's not this complex anymore, but IIRC the `np.min` returns a float so the `isclose` is there to assure there is a tolerance up to 1e-5.

Reply:
I know how `isclose` works, what I meant to ask was what are these lines testing?  Maybe I should have asked "what is `np.max(np.where(img)[0])` and why should it be `640`?"

looking again I can see that you're testing the bounds/limits of the non-zero parts of the image (how far in the x/y directions the smoothing kernel spreads the heatmap when the gaze was perfectly fixed on one central pixel).  The point I was making (in retrospect, probably too subtly) was that ideally it should be easy/obvious to see what each line of the test is doing. informative variable names can go a long way (e.g., something like `expected_x_limits_of_activity = (860, 1060)` or `nonzero_pixels_idx_horiz = np.where(img)[0]`).

NOTE: not saying you should bring back these lines if you've already found a better/simpler way to test. Just taking advantage of a "teachable moment" :)

Reply:
oh okay, sorry! 😅 

Fair advice, even I had to double take when looking back at it.. I've simplified the test such that I don't apply or test the smoothing kernel. Instead I check that the only non-zero coordinate on the heatmap is `(960, 540)`, since we simulated the gaze to be perfectly fixated at pixel `(960, 540)`.

Since we are using scipy for the smoothing, it felt like overkill to test the bounds of the smoothing kernel. i.e. if we test that (without any smoothing) the heatmap values accurately reflect the gaze coordinates we pass in, then if we _do_ apply smoothing, we can trust that `scipy.ndimage.gaussian_filter`  is doing its job?

Reply:
> if we test that (without any smoothing) the heatmap values accurately reflect the gaze coordinates we pass in, then if we _do_ apply smoothing, we can trust that `scipy.ndimage.gaussian_filter` is doing its job?

makes sense to me